docs/content/concepts/assets/software-defined-assets.mdx

```diff
@@ -65,6 +65,48 @@ def downstream_asset(upstream_asset):
 
 The [explicit dependencies](#explicit-dependencies) example covers an alternative way to specify asset dependencies without needing to match argument names to upstream asset names.
 
+### Graph-backed assets
+
+[Basic software-defined assets](#a-basic-software-defined-asset) can only produce one data artifact. If generating an asset involves multiple discrete computations, you can use graph-backed assets by separating each computation into an op and building a graph to combine your computations. This way, each discrete computation can be reused in other assets and jobs.
+
+This approach to asset creation allows you to define a graph of ops which can produce one or multiple assets. **Note**: To use graphs to create an asset, the graph **must return an output**.
+
+To define a graph-backed asset, use the `from_graph` attribute on the `AssetsDefinition` object:
+
+```python file=/concepts/assets/graph_backed_asset.py startafter=start example endbefore=end example
+@op(required_resource_keys={"slack"})
+def fetch_files_from_slack(context) -> DataFrame:
+    files = context.resources.slack.files_list(channel="#random")
+    return DataFrame(
+        [
+            {
+                "id": file.get("id"),
+                "created": file.get("created"),
+                "title": file.get("title"),
+                "permalink": file.get("permalink"),
+            }
+            for file in files
+        ]
+    )
+
+
+@op
+def store_files(files):
+    return files.to_sql(name="slack_files", con=create_db_connection())
+
+
+@graph
+def store_slack_files_in_sql():
+    store_files(fetch_files_from_slack())
+
+
+graph_asset = AssetsDefinition.from_graph(store_slack_files_in_sql)
+```
+
+**Note**: All output assets must be selected when using a graph-backed asset to create a job. Dagster will select all graph output automatically upon creating a job.
```

Comment:
@clairelin135 Is the second part of the example you provided (specifically, what came after the `# end example` comment) intended to be the example provided here?

```python
slack_mock = mock.MagicMock()

store_slack_files = AssetGroup(
    [graph_asset],
    resource_defs={"slack": ResourceDefinition.hardcoded_resource(slack_mock)},
).build_job("store_slack_files")
```

Reply:
Nope, this is for testing purposes. Though this will error because the the database connection doesn't exist.